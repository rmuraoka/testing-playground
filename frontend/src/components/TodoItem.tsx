import React, { useState } from 'react';
import { Box, Typography, TextField, Button } from '@mui/material';

export interface TodoItemProps {
    id: number;
    title: string;
    description: string;
    onUpdate: (id: number, title: string, description: string) => void;
    onDelete: (id: number) => void;
}

const TodoItem: React.FC<TodoItemProps> = ({ id, title, description, onUpdate, onDelete }) => {
    const [isEditing, setIsEditing] = useState(false);
    const [editedTitle, setEditedTitle] = useState(title);
    const [editedDescription, setEditedDescription] = useState(description);

    const handleEditClick = () => {
        setIsEditing(true); // 編集モードに切り替え
    };

    const handleSaveClick = () => {
        onUpdate(id, editedTitle, editedDescription); // 更新処理を呼び出し
        setIsEditing(false); // 編集モードを終了
    };

    const handleDeleteClick = (event: React.MouseEvent) => {
        event.stopPropagation();
        onDelete(id); // 削除処理を呼び出し
    };

    return (
        <Box p={2} border={1} borderRadius={4} mb={2}>
            {isEditing ? (
                <>
                    <TextField
                        label="Title"
                        variant="outlined"
                        value={editedTitle}
                        onChange={(e) => setEditedTitle(e.target.value)}
                    />
                    <TextField
                        label="Description"
                        variant="outlined"
                        value={editedDescription}
                        onChange={(e) => setEditedDescription(e.target.value)}
                    />
                    <Button onClick={handleSaveClick} variant="contained" color="primary">
                        Save
                    </Button>
                </>
            ) : (
                <>
                    <div onClick={handleEditClick}>
                        <Typography variant="h6">{title}</Typography>
                        <Typography variant="body2" color="textSecondary">
                            {description}
                        </Typography>
                        <Button data-testid={`delete-task-button-${title}`} onClick={handleDeleteClick} variant="outlined" color="secondary">
                            Delete
                        </Button>
                    </div>
                </>
            )}
        </Box>
    );
};

export default TodoItem;
