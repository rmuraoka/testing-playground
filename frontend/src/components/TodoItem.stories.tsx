import React from 'react';
import {Meta} from '@storybook/react';
import TodoItem from './TodoItem';

export default {
    title: 'TodoItem',
    component: TodoItem,
} as Meta;

export const Default = () =>
    <TodoItem
        id={1}
        title={'タイトル'}
        description={'本文'}
        onUpdate={() => {
            console.log('onUpdate');
        }}
    />;
