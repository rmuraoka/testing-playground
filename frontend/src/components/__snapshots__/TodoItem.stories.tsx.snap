// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Default smoke-test 1`] = `
<div class="MuiBox-root css-mkpezx">
  <div>
    <h6 class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root">
      タイトル
    </h6>
    <p class="MuiTypography-root MuiTypography-body2 css-1a1whku-MuiTypography-root">
      本文
    </p>
    <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary css-1x6kmiy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
            data-testid="delete-task-button-タイトル"
    >
      Delete
    </button>
  </div>
</div>
`;
